### PR TITLE
[GHA] Python test optimizations 

### PR DESCRIPTION
**Pull request by mryzhov**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
Optimization - do you mean split python jobs into 2 or several ones?
Do we need it for now? Maybe we need to optimize and remove repository clone for python tests? It will save us 2 minutes

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -741,6 +741,101 @@ jobs:
...
+    steps:
+      - name: Create Directories
+        run: |
+          mkdir -p ${{ env.INSTALL_DIR }} ${{ env.INSTALL_TEST_DIR }}
+
+      - name: Clone OpenVINO
+        uses: actions/checkout@v4
+        with:
+          path: 'openvino'
+          submodules: 'recursive'
```
Comment (ilya-lavrenov) [inline .github/workflows/linux.yml:766]: please, apply generic changes from here

File: .github/workflows/linux.yml [hunk 2]
```diff
@@ -741,6 +741,101 @@ jobs:
             ${{ env.INSTALL_TEST_DIR }}/TEST*.xml
           if-no-files-found: 'error'
 
+  Python_Unit_Tests_Model_Optimizer:
```
Comment (ilya-lavrenov) [inline .github/workflows/linux.yml:744]: why these tests are duplicated and not removed from common python job?

Comment (mryzhov) [reply .github/workflows/linux.yml:744]: because this is draft, please do not review for a while.

File: tools/mo/CMakeLists.txt [hunk 3]
```diff
@@ -22,6 +22,12 @@ configure_file("${CMAKE_CURRENT_SOURCE_DIR}/automation/version.txt.in"
 
 if(ENABLE_TESTS)
     install(FILES requirements_dev.txt
+                  requirements_caffe.txt
```
Comment (ilya-lavrenov) [inline tools/mo/CMakeLists.txt:25]: the main problem here is that these files refer to `-c ../contrainsts.txt` files, which is relative path. Once we install it into test package, this relative path is not valid anymore 😢

Comment (ilya-lavrenov) [reply tools/mo/CMakeLists.txt:25]: ok, then you need to install this dependency file (`constants.txt`) as well

Comment (mryzhov) [reply tools/mo/CMakeLists.txt:25]: yes , it is already there )

File: .github/workflows/linux.yml [hunk 4]
```diff
@@ -732,22 +673,22 @@ jobs:
...
+          PYTHONPATH: ${{ env.LAYER_TESTS_INSTALL_DIR }}:$PYTHONPATH
           TEST_DEVICE: CPU
           TEST_PRECISION: FP16
 
       - name: TensorFlow Lite Layer Tests - TFL FE
         run: |
           python3 -m pytest ${{ env.LAYER_TESTS_INSTALL_DIR }}/tensorflow_lite_tests/ --junitxml=${{ env.INSTALL_TEST_DIR }}/TEST-tfl_fe.xml
         env:
-          PYTHONPATH: ${{ env.INSTALL_DIR }}/tools/mo/:${{ env.LAYER_TESTS_INSTALL_DIR }}:$PYTHONPATH
+          PYTHONPATH: ${{ env.LAYER_TESTS_INSTALL_DIR }}:$PYTHONPATH
```
Comment (ilya-lavrenov) [inline .github/workflows/linux.yml:691]: by default PYTHONPATH is empty, so we don't need to extend it, we can just set.

**Comment by ilya-lavrenov** [comment]
Let's avoid split on multiple Jobs, I prefer to avoid only extra clone in Python Unit tests (if possible), plus clean-up of steps not to point to repo sources / setupvars.sh - better to run tests against wheel package

**Pull request closed without merging**